synapse/handlers/federation.py

```diff
@@ -1705,8 +1719,31 @@ async def _sync_partial_state_room_wrapper() -> None:
                     room_id=room_id,
                 )
             finally:
+                # Check whether the room is still partial stated, while we still claim
+                # to be the active sync. Usually, the partial state flag will be gone,
+                # unless we left and rejoined the room, or the sync failed.
+                is_still_partial_state_room = await self.store.is_partial_state_room(
+                    room_id
+                )
                 self._active_partial_state_syncs.remove(room_id)
 
+                # Check if we need to restart the sync.
+                if room_id in self._partial_state_syncs_to_restart:
+                    (
+                        restart_initial_destination,
+                        restart_other_destinations,
+                    ) = self._partial_state_syncs_to_restart[room_id]
+
+                    # Clear the restart flag.
+                    self._partial_state_syncs_to_restart.pop(room_id, None)
```

Comment:
I was going to say "Can't we combine these lines?", but I think this version is going to be more mypy-friendly(?)

Reply:
Yes we can 👍